### PR TITLE
fix(core): add outputs to WorkingDirectory (#3693)

### LINKED ISSUE #3693: WorkingDirectory has outputFiles property but it doesn't actually capture outputFiles
### Describe the issue

Reproducer: the dbt task should capture all files from the cloned Git directory as `outputFiles` property:

image

```yaml
id: git
namespace: tutorial

tasks:
  - id: dbt
    type: io.kestra.core.tasks.flows.WorkingDirectory
    outputFiles:
      - "**/*"
    tasks:
      - id: clone
        type: io.kestra.plugin.git.Clone
        url: https://github.com/kestra-io/dbt-example
        branch: main

  - id: list
    type: io.kestra.plugin.scripts.shell.Commands
    inputFiles: "{{ outputs.dbt.outputFiles }}"
    commands:
      - ls -R .
```

The log shows the files were captured:

image

However, it captures nothing in outputs:

image

image

### Environment

- Kestra Version:
- Operating System (OS/Docker/Kubernetes):
- Java Version (if you don't run kestra in Docker):

### PR DESCRIPTION
**Pull request by fhussonnois**
Fix: #3693

### CODE REVIEW

File: core/src/main/java/io/kestra/core/models/tasks/FlowableTask.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ default Optional<State.Type> resolveState(RunContext runContext, Execution execu
         );
     }
 
-    default T outputs(RunContext runContext, Execution execution, TaskRun parentTaskRun) throws IllegalVariableEvaluationException {
+    default T outputs(RunContext runContext) throws Exception {
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/models/tasks/FlowableTask.java:76]: Can we only throws IllegalVariableEvaluationException as before?

File: core/src/main/java/io/kestra/core/runners/ExecutorService.java
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/runners/ExecutorService.java:1]: You can check with Ludo if you prefere but if saveFlowableOutput seems to be useless let's remove it (in a separate commit so we can easily add it back)

**Review by loicmathieu — APPROVED** [review]
2 remarks but you can merge it anyway if you don't think they are important.

File: core/src/main/java/io/kestra/core/models/tasks/FlowableTask.java (hunk 1 above)
Comment (fhussonnois) [reply core/src/main/java/io/kestra/core/models/tasks/FlowableTask.java:76]: The outputs implementation for WorkingDirectory can throw an IOException and it's better to propagate the exception.

**Pull request merged**